DirectProgramming/C++SYCL_FPGA/Tutorials/Features/task_sequences/README.md

```diff
@@ -0,0 +1,249 @@
+|--------------------|----------|---------------|---
+| Task function      | callable | N/A           | A callable object `f` that defines the asynchronous task to be associated with the `task_sequence`. The callable object `f` must meet the following requirements: <br> • The object `f` must be statically resolvable at compile time, which means it is not a function pointer. <br> • The object `f` must not be an overloaded function. <br> • The return type (`ReturnT`) and argument types (`ArgsT…`) of object f must be resolvable and fixed.
+| Invocation Capacity* | `uint32_t` | 1 | The size of the hardware queue instantiated for `async()` function calls. This parameter value corresponds to the minimum number of outstanding `async()` function calls to be supported. When the outstanding number of `async()` function calls reach this value, further calls may block until the number of outstanding calls is reduced to the `invocation_capacity`. The default value of this parameter is 1.
+| Response Capacity* | `uint32_t` | 1 | The size of the hardware queue instantiated to hold task function results. This parameter value corresponds to the maximum number of outstanding `async()` calls such that all outstanding tasks are guaranteed to make forward progress. Further `async()` calls may block until the number of outstanding calls reduce to the `response_capacity`. The default value of this parameter is 1.
+
+The following example shows how to use task Sequence.  
+
+```c++
+// Task function
+int mult(int a, int b) {
+	 return a * b;
+}
+
+int mult_and_add(int a1, int b1, int a2, int b2) {
+  task_sequence<mult> task1, task2;
+  task1.async(a1, b1);
+  task2.async(a2, b2);
+  return task1.get() + task2.get();
+}
+```
+C++ function, mult is the task function with asynchronous activity defined. The task function is used to parameterize the task_sequence class with two object instances, task1 and task2. Parallel invocations of the task function is then launched through async() method of each object instance and their results collected by calls to the get() method on each object. The
+async() function call is non-blocking, and it returns before the asynchronous f invocation completes executing and potentially before f even begins executing, as the return type from the async() function provides no implicit information on the execution status of f.
+The get() function retrieves the oldest result from this logical FIFO queue and blocks (waits) until a result is available if no result is available immediately upon the call to the get() function. The return type of the get() function is the same as the return type of the task function .
+
+You can see concrete example of kernel that use task sequences to run 4 loops concurrently in `task_sequence/main.cpp` and naively running the 4 loops sequentially in naive method in `naive/main.cpp`.
+
+## Sample Structure
+The 2 different example designs in this sample perform similar operations.
```

Comment:
```suggestion
The 2 different example designs in this sample perform similar operations.
```
```suggestion
The 2 different example designs in this sample perform similar operations. You may compare the C++ source files to see the code changes that are necessary to convert a design to use task sequences instead of sequential loops.
```